sycl/include/CL/sycl/INTEL/intel_matrix/matrix-amx.hpp

```diff
@@ -0,0 +1,427 @@
+//===-------------- matrix-amx.hpp - SYCL matrix --------------*- C++ -*---===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+///
+/// We provide new interfaces for matrix muliply in this patch:
+/// 1. A new class called joint_matrix is introduced, and the user needs to
+/// specify the type of the elements, sizes, and the memory layout.
+///
+/// 2. joint_matrix_load is used for loading data from main memory to tiles of
+/// AMX or kernel's local memory.
+///
+/// 3. joint_matrix_store is used for storing data tiles of AMX or kernel's
+/// local memory to main memory.
+///
+/// 4. joint_matrix_mad is used for the matrix multiply and add function.
+/// It performs the multiply operation on the matrices A and B, accumulates the
+/// result with C and returns the result.
+///
+/// The following operation can be realized with the interfaces:
+///  C = A*B+C
+/// 1. All cases where A(int8, any-size, row_major), B(int8, any-size,
+/// packed_b), C(int32, any-size, row_major)
+/// 2. All cases where A(bf16, any-size, row_major), B(bf16, any-size,
+/// packed_b), C(float, any-size, row_major)
+///
+///
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
```

Comment:
Please, add following include defining `__SYCL_INLINE_NAMESPACE` and `__SYCL_ALWAYS_INLINE`.
```suggestion

#include <CL/sycl/detail/defines_elementary.hpp>
```